.golangci.yml

```diff
@@ -108,3 +108,8 @@ issues:
     - path: _test\.go
       linters:
         - scopelint
+    # Exclude some staticcheck messages
+    - linters:
+      - staticcheck
+      # See https://github.com/golang/protobuf/issues/1077
```

Comment:
Protobuf v1.4.2 has `// Deprecated` comment in
https://github.com/golang/protobuf/blob/d04d7b157bb510b1e0c10132224b616ac0e26b17/proto/proto.go#L12
which was introduced in https://go-review.googlesource.com/c/protobuf/+/213901

The change below completely disables the following check:
```
SA1019 | Using a deprecated function, variable, constant or field
```
Is there any better workaround?

Reply:
AFAIU migrating to the package google.golang.org/protobuf/proto will require re-generating protobuf types, some of which live outside of this repo (e.g. Zipkin, OpenCensus)

Reply:
I think you can change OpenCensus. We need to also see if Zipkin has a new release which uses the new libreary

Reply:
Is it possible to do a line based ignore on the import statement? https://staticcheck.io/docs/#line-based-linter-directives

Reply:
Do we still need this?

Reply:
I've reverted the blanket ignore rule in the `STATICCHECK` Makefile rule, but left this one.
It seems that for some reason we run `staticcheck` twice and they conflict with each other when I try to add ignore rules:
- We have a separate target for `staticcheck`: https://github.com/open-telemetry/opentelemetry-collector/blob/c4a97db078c04ee33d12016f9df6bea6872bacaa/Makefile#L111. To ignore errors which can't be trivially fixed, I added `//lint:ignore SA1019`
- We also run `staticcheck` as part of `golangci-lint`, see https://github.com/open-telemetry/opentelemetry-collector/blob/c4a97db078c04ee33d12016f9df6bea6872bacaa/.golangci.yml#L100. It requires `//nolint:staticcheck` comment which conflicts with the one above (only left-most comment disables linter)

@bogdandrutu Why do we run `staticcheck` twice?
For now I kept the `SA1019` check in the `lint-static-check`, but excluded it from `golangci-lint`.
We can deduplicate running `staticcheck` in a separate PR if needed.